server/protocols/protocols-lmtp/src/main/java/org/apache/james/lmtpserver/hook/MailboxDeliverToRecipientHandler.java

```diff
@@ -90,13 +90,13 @@ public HookResult deliver(SMTPSession session, MailAddress recipient, MailEnvelo
             result = new HookResult(HookReturnCode.OK, SMTPRetCode.MAIL_OK, DSNStatus.getStatus(DSNStatus.SUCCESS, DSNStatus.CONTENT_OTHER) + " Message received");
 
         } catch (IOException e) {
-            session.getLogger().info("Unexpected error handling DATA stream", e);

```

Comment:
Can't you factorize all those catch: `catch(IOException | MailboxException | UsersRepositoryException e)`